### ISSUE TITLE
Facebook-like tagging

### ISSUE THREAD

**Issue by cmer** [alexgorbatchev/jquery-textext#12]
I am trying to replicate Facebook's in-comment tagging capability. Basically, when you start typing a name, it triggers an Ajax search. It waits for the first 2 characters (1 upper case and 1 lower case) to do so and shows name suggestions.

The closest example I've found to this is http://textextjs.com/manual/examples/autocomplete-with-tags.html but it turns everything into a tag on Enter, which isn't the expected behavior in this case. I want to be able to do:

"Hey [Alex Gorbatchev], check this out!"

http://textextjs.com/manual/examples/ajax-with-filter-tags-and-autocomplete.html is another close example of what I want to achieve, but it rejects everything that is not in the tag list.

Is there a way to accomplish what I'm looking with this plugin? Also, is there a way to know what was a tag and what wasn't on the "server side"? From what I understand it will just send a simple string back. I'd like to know, for example, that "Alex Gorbatchev" was tagged on the client-side.

Also, in the autocomplete-with-tags example, clicking on the suggestion in the drop-down does not auto-complete the item. I'm not sure if that's a bug. I will file a separate issue if it is.

**Comment by alexgorbatchev** [comment]
Could you post a screenshot of what you expect? If I understand you correctly, you want to have a tag in the middle of a plain custom text. It sounds like an awesome feature, but is currently not supported.

**Comment by cmer** [comment]
Yes exactly what I want! For example: https://skitch.com/cmer/g5f2s/6-facebook

In a world where people tag other people in comments and such, it would indeed be an awesome feature!

What would it take to implement this?

**Comment by alexgorbatchev** [comment]
Where on facebook can I see this? Can't seem to find this case.

**Comment by cmer** [comment]
Just comment on something on Facebook and start typing the name of somebody else who commented (not sure if that person has to be your friend or not). Don't forget to start the name with a cap. It'll show an autocomplete list.

**Comment by alexgorbatchev** [comment]
I see what you mean. It's looks more like looser version of autocomplete.

**Comment by cmer** [comment]
How hard would it be to accomplish that what your plugin?

**Comment by alexgorbatchev** [comment]
I feel that this feature should go into the autocomplete plugin. It's would be a fair bit of work, primarily because currently there's no way to highlight portion of the text in the input box.

**Comment by afeld** [comment]
I'd like to take a whack at this... what is the status of the `development` branch?  Should I start from there, or just stick with `master`?

**Comment by alexgorbatchev** [comment]
@afeld `development` is the most recent work. It's somewhat of a major rework actually. I think it's mostly stable, but documentation is horribly out of date. If you will be working on it, lets get in touch over gchat and I'd love to collaborate.

**Comment by afeld** [comment]
Cool, my gmail address is on my [profile](https://github.com/afeld).  Thanks!

**Comment by afeld** [comment]
See also: https://github.com/willywos/bangbang

**Comment by mrbonezy** [comment]
@cmer any conclusions on how to achieve this?

**Comment by cmer** [comment]
We ended up building our own. We couldn't find anything that worked for us...

**Comment by mrbonezy** [comment]
Have you tried this one? After extensive search it seems like the only one that does what I need
https://github.com/podio/jquery-mentions-input
